### PR TITLE
Fixing CVE issues by bumping GO and Text versions.

### PR DESCRIPTION
**Pull request by atshaw43**
*Issue #, if available:*
Internally tracked

*Description of changes:*
Bumping GO to version 1.18.
Bumping golang.org/x/text to v0.3.7 
Bumping golang.org/x/net to v0.0.0-20220809184613-07c6da5e1ced
Bumping golang.org/x/sys to v0.0.0-20220808155132-1c4a2a72c664

*Testing*
Ran 'make test'.
Ran the code in Docker. Validated that the container was running.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -25,14 +25,11 @@ golang.org/x/net v0.0.0-20200202094626-16171245cfb2/go.mod h1:z5CRVTTTmAJ677TzLL
 golang.org/x/net v0.0.0-20210716203947-853a461950ff h1:j2EK/QoxYNBsXI4R7fQkkRUk8y6wnOBI+6hgPdP/6Ds=
```
Comment (atshaw43) [inline go.sum:25]: This file is auto-generated.

File: go.mod [hunk 2]
```diff
@@ -1,15 +1,23 @@
 module github.com/aws/aws-xray-daemon
```
Comment (atshaw43) [inline go.mod:1]: I changed the GO and golang.org/x/text version manually. The other changes were auto-generated by go mod tidy. That includes the additional require block.

**Comment by atshaw43** [comment]
>>> aws-xray-daemon % make test
Testing daemon
go test -cover ./...
?   	github.com/aws/aws-xray-daemon/cmd/tracing	[no test files]
ok  	github.com/aws/aws-xray-daemon/pkg/bufferpool	1.313s	coverage: 100.0% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/cfg	5.526s	coverage: 57.6% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/cli	1.031s	coverage: 80.0% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/conn	5.466s	coverage: 67.9% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/logger	2.257s	coverage: 33.3% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/processor	2.698s	coverage: 84.9% of statements
?   	github.com/aws/aws-xray-daemon/pkg/profiler	[no test files]
ok  	github.com/aws/aws-xray-daemon/pkg/proxy	4.521s	coverage: 70.4% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/ringbuffer	5.486s	coverage: 85.7% of statements
?   	github.com/aws/aws-xray-daemon/pkg/socketconn	[no test files]
?   	github.com/aws/aws-xray-daemon/pkg/socketconn/udp	[no test files]
ok  	github.com/aws/aws-xray-daemon/pkg/telemetry	5.376s	coverage: 18.4% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/tracesegment	1.392s	coverage: 88.9% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/util	1.875s	coverage: 96.3% of statements
ok  	github.com/aws/aws-xray-daemon/pkg/util/test	3.383s	coverage: 77.1% of statements
?   	github.com/aws/aws-xray-daemon/pkg/util/timer	[no test files]

File: go.mod [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 
-go 1.16
+go 1.18
```
Comment (willarmiros) [inline go.mod:3]: In order to affect the version of Golang used in the Docker images, we need to update the version of the Golang base image in the Dockerfiles:
1. https://github.com/aws/aws-xray-daemon/blob/master/Dockerfile.amazonlinux#L2
2. https://github.com/aws/aws-xray-daemon/blob/master/Dockerfile#L2

And we should bump the version of Go we use in the CI/CD too probably, e.g. https://github.com/aws/aws-xray-daemon/blob/dd6aef5fc1b15ca4afecc93b491fd267d3c4eded/.github/workflows/release-build-publish.yml#L25

File: go.mod [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
 
 require (
-	github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d // indirect
 	github.com/aws/aws-sdk-go v1.29.29
 	github.com/cihub/seelog v0.0.0-20170130134532-f561c5e57575
-	github.com/go-ole/go-ole v1.2.4 // indirect
 	github.com/shirou/gopsutil v2.19.10+incompatible
 	github.com/stretchr/testify v1.4.0
 	golang.org/x/net v0.0.0-20210716203947-853a461950ff
 	golang.org/x/sys v0.0.0-20210423082822-04245dca01da
```
Comment (willarmiros) [inline go.mod:11]: Can we bump the net and sys packages to latest as well?

File: go.mod (hunk 3 above)
Comment (atshaw43) [reply go.mod:3]: Good catch. That was off my radar.

**Comment by atshaw43** [comment]
image
Docker runs after upgrading the version.

File: go.mod (hunk 4 above)
Comment (atshaw43) [reply go.mod]: Done

**Review by willarmiros — APPROVED** [review]
lgtm thanks!

**Pull request merged**